### ISSUE TITLE
Yolov8 Mlflow integration on Azure Databricks

### ISSUE THREAD

**Issue by DeTandtThibaut** [ultralytics/ultralytics#2238]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Integrations

### Bug

When using Yolov8 in Databricks it gives the following error. Yolo tries to look for a look for a non-existent experiment.
Databricks has it's own mlflow implementation and seems to be colliding with the mlflow integration from yolo, resulting in a bad request without additional context, which was a pain in the as to debug
![MicrosoftTeams-image (7)](https://user-images.githubusercontent.com/33635345/234286963-95ed11dd-b8d4-44ef-bd93-90d14b3d24fe.png)

### Environment

Ultralytics YOLOv8.0.87 🚀 Python-3.9.5 torch-1.13.1+cpu CPU

### Minimal Reproducible Example

```
# You only live once, right?

from ultralytics import YOLO
import mlflow


mlflow.autolog()

mlflow.start_run()

model = YOLO('yolov8n-seg.pt')  # load a pretrained model (recommended for training)

# Use the model
results = model.train(data='/dbfs/mnt/factoryImages/model/yolo/segmentation/latest__segmentation_labels/latest__segmentation_labels.yaml', epochs=100, project= '../../tmp' ,name='segmentation', patience = 4)  # train the model
results = model.val()  # evaluate model performance on the validation set

mlflow.end_run()
```

### Additional

this is a conflict between databricks & yolov8

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by AyushExel** [comment]
Thanks for bringing this up. We can't reproduce Azure/databricks specific errors as we don't have access to such systems. But if you'd like to contribute a fix, we can guide you with that. Thanks!

**Comment by Toutatis64** [comment]
│ /local_disk0/.ephemeral_nfs/envs/pythonEnv-40ed9144-e47f-43f6-9aa9-c570c42e5 │
│ 866/lib/python3.9/site-packages/ultralytics/yolo/engine/trainer.py:166 in    │
│ run_callbacks                                                                │
│                                                                              │
│   163 │   def run_callbacks(self, event: str):                               │
│   164 │   │   """Run all existing callbacks associated with a particular eve │
│   165 │   │   for callback in self.callbacks.get(event, []):                 │
│ ❱ 166 │   │   │   callback(self)                                             │
│   167 │                                                                      │
│   168 │   def train(self):                                                   │
│   169 │   │   """Allow device='', device=None on Multi-GPU systems to defaul │
│                                                                              │
│ ╭───────────────────────────────── locals ─────────────────────────────────╮ │
│ │ callback = <function on_train_end at 0x7f991b429160>                     │ │
│ │    event = 'on_train_end'                                                │ │
│ │     self = <ultralytics.yolo.v8.detect.train.DetectionTrainer object at  │ │
│ │            0x7f9868293040>                                               │ │
│ ╰──────────────────────────────────────────────────────────────────────────╯ │
│                                                                              │
│ /local_disk0/.ephemeral_nfs/envs/pythonEnv-40ed9144-e47f-43f6-9aa9-c570c42e5 │
│ 866/lib/python3.9/site-packages/ultralytics/yolo/utils/callbacks/mlflow.py:6 │
│ 7 in on_train_end                                                            │
│                                                                              │
│   64 │   │   root_dir = Path(__file__).resolve().parents[3]                  │
│   65 │   │   run.log_artifact(trainer.best)                                  │
│   66 │   │   model_uri = f'runs:/{run_id}/'                                  │
│ ❱ 67 │   │   run.register_model(model_uri, experiment_name)                  │
│   68 │   │   run.pyfunc.log_model(artifact_path=experiment_name,             │
│   69 │   │   │   │   │   │   │    code_path=[str(root_dir)],                 │
│   70 │   │   │   │   │   │   │    artifacts={'model_path': str(trainer.save_ │
│                                                                              │
│ ╭───────────────────────────────── locals ─────────────────────────────────╮ │
│ │ model_uri = 'runs:/8636e556b837409ba7fc0745a7307e18/'                    │ │
│ │  root_dir = PosixPath('/local_disk0/.ephemeral_nfs/envs/pythonEnv-40ed9… │ │
│ │   trainer = <ultralytics.yolo.v8.detect.train.DetectionTrainer object at │ │
│ │             0x7f9868293040>                                              │ │
│ ╰──────────────────────────────────────────────────────────────────────────╯ │
│                                                                              │
│ /local_disk0/.ephemeral_nfs/envs/pythonEnv-40ed9144-e47f-43f6-9aa9-c570c42e5 │
│ 866/lib/python3.9/site-packages/mlflow/tracking/_model_registry/fluent.py:74 │
│  in register_model                                                           │
│                                                                              │
│   71 │   │   │   │   % name                                                  │
│   72 │   │   │   )                                                           │
│   73 │   │   else:                                                           │
│ ❱ 74 │   │   │   raise e                                                     │
│   75 │                                                                       │
│   76 │   if RunsArtifactRepository.is_runs_uri(model_uri):                   │
│   77 │   │   source = RunsArtifactRepository.get_underlying_uri(model_uri)   │
│                                                                              │
│ ╭───────────────────────────────── locals ─────────────────────────────────╮ │
│ │ await_registration_for = 300                                             │ │
│ │                 client = <mlflow.tracking.client.MlflowClient object at  │ │
│ │                          0x7f974ecfe8e0>                                 │ │
│ │              model_uri = 'runs:/8636e556b837409ba7fc0745a7307e18/'       │ │
│ │                   name = '/Shared/YOLOv8'                                │ │
│ │                   tags = None                                            │ │
│ ╰──────────────────────────────────────────────────────────────────────────╯ │
│                                                                              │
│ /local_disk0/.ephemeral_nfs/envs/pythonEnv-40ed9144-e47f-43f6-9aa9-c570c42e5 │
│ 866/lib/python3.9/site-packages/mlflow/tracking/_model_registry/fluent.py:65 │
│  in register_model                                                           │
│                                                                              │
│   62 │   """                                                                 │
│   63 │   client = MlflowClient()                                             │
│   64 │   try:                                                                │
│ ❱ 65 │   │   create_model_response = client.create_registered_model(name)    │
│   66 │   │   eprint("Successfully registered model '%s'." % create_model_res │
│   67 │   except MlflowException as e:                                        │
│   68 │   │   if e.error_code == ErrorCode.Name(RESOURCE_ALREADY_EXISTS):     │
│                                                                              │
│ ╭───────────────────────────────── locals ─────────────────────────────────╮ │
│ │ await_registration_for = 300                                             │ │
│ │                 client = <mlflow.tracking.client.MlflowClient object at  │ │
│ │                          0x7f974ecfe8e0>                                 │ │
│ │              model_uri = 'runs:/8636e556b837409ba7fc0745a7307e18/'       │ │
│ │                   name = '/Shared/YOLOv8'                                │ │
│ │                   tags = None                                            │ │
│ ╰──────────────────────────────────────────────────────────────────────────╯ │
│                                                                              │
│ /local_disk0/.ephemeral_nfs/envs/pythonEnv-40ed9144-e47f-43f6-9aa9-c570c42e5 │
│ 866/lib/python3.9/site-packages/mlflow/tracking/client.py:1847 in            │
│ create_registered_model                                                      │
│                                                                              │
│   1844 │   │   │   tags: {'nlp.framework': 'Spark NLP'}                      │
│   1845 │   │   │   description: This sentiment analysis model classifies the │
│   1846 │   │   """                                                           │
│ ❱ 1847 │   │   return self._get_registry_client().create_registered_model(na │
│   1848 │                                                                     │
│   1849 │   def rename_registered_model(self, name: str, new_name: str) -> Re │
│   1850 │   │   """                                                           │
│                                                                              │
│ ╭───────────────────────────────── locals ─────────────────────────────────╮ │
│ │ description = None                                                       │ │
│ │        name = '/Shared/YOLOv8'                                           │ │
│ │        self = <mlflow.tracking.client.MlflowClient object at             │ │
│ │               0x7f974ecfe8e0>                                            │ │
│ │        tags = None                                                       │ │
│ ╰──────────────────────────────────────────────────────────────────────────╯ │
│                                                                              │
│ /local_disk0/.ephemeral_nfs/envs/pythonEnv-40ed9144-e47f-43f6-9aa9-c570c42e5 │
│ 866/lib/python3.9/site-packages/mlflow/tracking/_model_registry/client.py:59 │
│  in create_registered_model                                                  │
│                                                                              │
│    56 │   │   #       Those are constraints applicable to any backend, given │
│    57 │   │   tags = tags if tags else {}                                    │
│    58 │   │   tags = [RegisteredModelTag(key, str(value)) for key, value in  │
│ ❱  59 │   │   return self.store.create_registered_model(name, tags, descript │
│    60 │                                                                      │
│    61 │   def update_registered_model(self, name, description):              │
│    62 │   │   """                                                            │
│                                                                              │
│ ╭───────────────────────────────── locals ─────────────────────────────────╮ │
│ │ description = None                                                       │ │
│ │        name = '/Shared/YOLOv8'                                           │ │
│ │        self = <mlflow.tracking._model_registry.client.ModelRegistryClie… │ │
│ │               object at 0x7f974ecfea00>                                  │ │
│ │        tags = []                                                         │ │
│ ╰──────────────────────────────────────────────────────────────────────────╯ │
│                                                                              │
│ /local_disk0/.ephemeral_nfs/envs/pythonEnv-40ed9144-e47f-43f6-9aa9-c570c42e5 │
│ 866/lib/python3.9/site-packages/mlflow/store/model_registry/rest_store.py:83 │
│  in create_registered_model                                                  │
│                                                                              │
│    80 │   │   req_body = message_to_json(                                    │
│    81 │   │   │   CreateRegisteredModel(name=name, tags=proto_tags, descript │
│    82 │   │   )                                                              │
│ ❱  83 │   │   response_proto = self._call_endpoint(CreateRegisteredModel, re │
│    84 │   │   return RegisteredModel.from_proto(response_proto.registered_mo │
│    85 │                                                                      │
│    86 │   def update_registered_model(self, name, description):              │
│                                                                              │
│ ╭───────────────────────────────── locals ─────────────────────────────────╮ │
│ │ description = None                                                       │ │
│ │        name = '/Shared/YOLOv8'                                           │ │
│ │  proto_tags = []                                                         │ │
│ │    req_body = '{\n  "name": "/Shared/YOLOv8"\n}'                         │ │
│ │        self = <mlflow.store.model_registry.rest_store.RestStore object   │ │
│ │               at 0x7f974ef17d90>                                         │ │
│ │        tags = []                                                         │ │
│ ╰──────────────────────────────────────────────────────────────────────────╯ │
│                                                                              │
│ /local_disk0/.ephemeral_nfs/envs/pythonEnv-40ed9144-e47f-43f6-9aa9-c570c42e5 │
│ 866/lib/python3.9/site-packages/mlflow/store/model_registry/rest_store.py:64 │
│  in _call_endpoint                                                           │
│                                                                              │
│    61 │   │   │   return call_endpoints(self.get_host_creds(), endpoints, js │
│    62 │   │   else:                                                          │
│    63 │   │   │   endpoint, method = _METHOD_TO_INFO[api]                    │
│ ❱  64 │   │   │   return call_endpoint(self.get_host_creds(), endpoint, meth │
│    65 │                                                                      │
│    66 │   # CRUD API for RegisteredModel objects                             │
│    67                                                                        │
│                                                                              │
│ ╭───────────────────────────────── locals ─────────────────────────────────╮ │
│ │                api = <class 'model_registry_pb2.CreateRegisteredModel'>  │ │
│ │ call_all_endpoints = False                                               │ │
│ │           endpoint = '/api/2.0/preview/mlflow/registered-models/create'  │ │
│ │          json_body = '{\n  "name": "/Shared/YOLOv8"\n}'                  │ │
│ │             method = 'POST'                                              │ │
│ │     response_proto = <class 'rich.pretty.Node'>.__repr__ returned empty  │ │
│ │                      string                                              │ │
│ │               self = <mlflow.store.model_registry.rest_store.RestStore   │ │
│ │                      object at 0x7f974ef17d90>                           │ │
│ ╰──────────────────────────────────────────────────────────────────────────╯ │
│                                                                              │
│ /local_disk0/.ephemeral_nfs/envs/pythonEnv-40ed9144-e47f-43f6-9aa9-c570c42e5 │
│ 866/lib/python3.9/site-packages/mlflow/utils/rest_utils.py:280 in            │
│ call_endpoint                                                                │
│                                                                              │
│   277 │   │   response = http_request(                                       │
│   278 │   │   │   host_creds=host_creds, endpoint=endpoint, method=method, j │
│   279 │   │   )                                                              │
│ ❱ 280 │   response = verify_rest_response(response, endpoint)                │
│   281 │   js_dict = json.loads(response.text)                                │
│   282 │   parse_dict(js_dict=js_dict, message=response_proto)                │
│   283 │   return response_proto                                              │
│                                                                              │
│ ╭───────────────────────────────── locals ─────────────────────────────────╮ │
│ │       endpoint = '/api/2.0/preview/mlflow/registered-models/create'      │ │
│ │     host_creds = <mlflow.utils.rest_utils.MlflowHostCreds object at      │ │
│ │                  0x7f974eac3340>                                         │ │
│ │      json_body = {'name': '/Shared/YOLOv8'}                              │ │
│ │         method = 'POST'                                                  │ │
│ │       response = <Response [400]>                                        │ │
│ │ response_proto = <class 'rich.pretty.Node'>.__repr__ returned empty      │ │
│ │                  string                                                  │ │
│ ╰──────────────────────────────────────────────────────────────────────────╯ │
│                                                                              │
│ /local_disk0/.ephemeral_nfs/envs/pythonEnv-40ed9144-e47f-43f6-9aa9-c570c42e5 │
│ 866/lib/python3.9/site-packages/mlflow/utils/rest_utils.py:206 in            │
│ verify_rest_response                                                         │
│                                                                              │
│   203 │   """Verify the return code and format, raise exception if the reque │
│   204 │   if response.status_code != 200:                                    │
│   205 │   │   if _can_parse_as_json_object(response.text):                   │
│ ❱ 206 │   │   │   raise RestException(json.loads(response.text))             │
│   207 │   │   else:                                                          │
│   208 │   │   │   base_msg = "API request to endpoint %s failed with error c │
│   209 │   │   │   │   endpoint,                                              │
│                                                                              │
│ ╭─────────────────────────── locals ────────────────────────────╮            │
│ │ endpoint = '/api/2.0/preview/mlflow/registered-models/create' │            │
│ │ response = <Response [400]>                                   │            │
│ ╰───────────────────────────────────────────────────────────────╯            │
╰──────────────────────────────────────────────────────────────────────────────╯
RestException: INVALID_PARAMETER_VALUE: Invalid name '/Shared/YOLOv8'. A

**Comment by Toutatis64** [comment]
I've also an issue with databricks due to the experiment_name that is linked to project and bad format name. I'm trying to solve that on fork and i'll push a PR if corrected. 
Otherwise, i had a workaround to disable the callback, which will be another PR ... (because today no way to disable mlflow callbacks)

**Comment by glenn-jocher** [comment]
@DeTandtThibaut it sounds like you have encountered some issues with the experiment name format in Databricks, and you are working on a solution by creating a fork and submitting a pull request. You have also found a workaround to disable the mlflow callbacks, which will also be submitted as a separate pull request. While these issues can be frustrating, it's great to see that you are actively working on solutions and contributing to the development community. Keep up the good work.

**Comment by yhamedish** [comment]
Im facing the same issue. Could you please clarify how do you disable the mlflow callback in databricks?

**Comment by Toutatis64** [comment]
Hi @yhamedish , you can disable mlflow callbacks in databricks using the pull request https://github.com/ultralytics/ultralytics/pull/2287 , where you have some example of code to disable it. Not yet integrated, but it works on databricks.

**Comment by glenn-jocher** [comment]
Hello @yhamedish, to disable mlflow callbacks in Databricks, you can refer to the pull request #2287 on the Ultralytics GitHub repository. This pull request provides an example code snippet that can be used to disable the mlflow callbacks. Although this pull request has not yet been integrated, it has been tested and proven to work on Databricks. I hope this helps!

**Comment by glenn-jocher** [comment]
@Toutatis64 @DeTandtThibaut @yhamedish great news 😃! Your original issue may now be resolved ✅ in PR #3790 and published in a new `ultralytics 8.0.141` release. This PR adds a SettingsManager class that allows for direct viewing and modifying of Ultralytics settings:

<img width="955" alt="Screenshot 2023-07-23 at 16 11 23" src="https://github.com/ultralytics/ultralytics/assets/26833433/1427f5bd-52b6-445a-ab5b-e4c0eeccc10b">

For details see https://docs.ultralytics.com/quickstart

To get this update:
- **[Git](https://github.com/ultralytics/ultralytics)** – Run `git pull` from within your `ultralytics/` directory or run `git clone https://github.com/ultralytics/ultralytics` again
- **[Pip](https://pypi.org/project/ultralytics/)** – Update with `pip install -U ultralytics`
- **[Notebooks](https://github.com/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb)** – Check out the updated notebooks  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/ultralytics)** – Run `sudo docker pull ultralytics/ultralytics:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and letting us know. Please confirm if this update fixes the issue for you, and don't hesitate to report any other **issues** you find or **feature requests** you may have. Happy training with YOLOv8 🚀!
